core/src/main/java/org/apache/iceberg/util/SortOrderUtil.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.util;
+
+import java.util.Collection;
+import org.apache.iceberg.PartitionField;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SortField;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.expressions.Expressions;
+import org.apache.iceberg.relocated.com.google.common.collect.Multimap;
+import org.apache.iceberg.relocated.com.google.common.collect.Multimaps;
+import org.apache.iceberg.transforms.SortOrderVisitor;
+
+public class SortOrderUtil {
+
+  private SortOrderUtil() {
+  }
+
+  public static SortOrder buildSortOrder(Table table) {
+    return buildSortOrder(table.spec(), table.sortOrder());
+  }
+
+  public static SortOrder buildSortOrder(PartitionSpec spec, SortOrder sortOrder) {
+    if (sortOrder.isUnsorted() && spec.isUnpartitioned()) {
```

Comment:
I return `sortOrder` if no sort order and the spec is unpartitioned. Otherwise, we will try to build an unsorted order with id = 1, which is not allowed.

Reply:
Should this return `SortOrder.unsorted()` instead?

Reply:
That should work too. Will update.

Reply:
Done. Resolving this.